source/concepts/applications.rst

```diff
@@ -0,0 +1,569 @@
+application to share where to display the molecular system relative to the
+users, and how to render the molecules.
+
+Frames
+~~~~~~
+
+The :ref:`trajectory service <trajectory-service>` allows to stream snapshots
+of arbitrary data to clients. Each snapshot is described in a :ref:`FrameData
+<frame-description>` which contains a key-value map of protobuf `Values
+<https://protobuf.dev/reference/protobuf/google.protobuf/#value>`_ and one of
+homogeneous arrays. Here, we define a set of keys and data format to describe
+the semantics of molecular systems.
+
+.. note::
+
+   A server using this set of keys can implement keys from another application
+   as well. For instance, a server implementing the :ref:`iMD application
+   <imd-application>` can implement both this set of keys and :ref:`iMD-specific
+   keys <imd-framedata-keys>`.
+
+All FrameData values used by the trajectory application use the following set
+of units:
+
+* lengths are expressed in nanometers (:math:`\text{nm}`)
+* durations are expressed in picoseconds (:math:`\text{ps}`)
+* masses are expressed in atomic mass unit (AMU)
+* charges are expressed in proton charge
+* energies are expressed in :math:`\text{kJ}\cdot\text{mol}^{-1}`
+* velocities are expressed in :math:`\text{nm}\cdot{ps}^{-1}`
+* forces are expressed in :math:`\text{kJ}\cdot\text{mol}^{-1}\cdot\text{nm}^{-1}`
+
+The coordinate system is the right-handed, Z-up, system used in most software
+working with molecular systems.
+
+Particles
+^^^^^^^^^
+
+A molecular system is composed of atoms. The application refers to them as
+"particles" to account for representations that do not deal with individual
+atoms, such as coarse-grained models (`e.g.` `Martini <http://cgmartini.nl/>`_
+or `SIRAH <http://www.sirahff.com/>`_). Particles are described by the following
+keys in the array map:
+
+* ``particle.positions``: the Cartesian coordinates of each particle. The
+  coordinates are stored as a flat array of coordinates where each triplet
+  corresponds to the XYZ coordinates of a particle.
+* ``particle.velocities``: the velocity of each particle. Like the positions,
+  they are expressed as a flattened array of triplets.
+* ``particle.forces``: the force array applied to each particle, as a flattened
+  array of triplets.
+* ``particle.elements``: the chemical element for each particle expressed as
+  atomic numbers. If a particle is not an atom, or if a chemical element is not
+  relevant for any reason, the atomic number can be set to 0.
+* ``particle.names``: a name for each particle. Each name is an arbitrary string
+  to identify the particle, usually within a residue. If an atom does not have
+  a name, set it to an empty string. When applicable, it is recommended to use
+  the names used in the Protein Data Bank.
+
+.. warning::
+
+   Many molecular dynamics integrators are based on the leap frog integration
+   method that calculates the velocities at the half step. Simulation engines
```

Comment:
```suggestion
   method that calculates the velocities at the half time step (e.g. at time :math:`t + \frac{\Delta t}{2}`). Simulation engines
```
Could it ever be true that the velocities are calculated for half a tilmestep behind the data being reported? Should the + be a \pm above?

Reply:
There is a potential ambiguity here: the formula describing the integrator says $t + \frac{\Delta t}{2}$ because $t$ is the step you start from and $t + \Delta t$ is the step you calculate. However, the frame contains the frame that has been calculated already, so if $t$ is the time for the frame in the `FrameData`, then the velocity is at $t - \frac{\Delta t}{2}$.

Reply:
Ah yes, I see, my mistake! So should the + in fact be a - in the suggestion?

Reply:
Or alternatively, should the example be abandoned?

Reply:
Agreed. Is this any better? For me it feels sensible to clarify what half step means as unambiguously as possible.
```suggestion
   method that calculates the velocities at the half step (i.e. half a time step out of sync with the positions). Simulation engines
```